src/compas_timber/planning/label.py

```diff
@@ -0,0 +1,157 @@
+from compas.data import Data
+
+from compas_timber.fabrication import AlignmentType
+from compas_timber.fabrication import Text
+
+
+class Label(object):
+    """A label for a timber assembly.
+
+    Attributes
+    ----------
+    element : :class:`compas_timber.structure.Beam`
+        The element to which the label is attached.
+    text : str
+        The text of the label.
+    """
+
+    def __init__(self, element, text=None):
+        self.element = element
+        self.user_text = text
+
+    @property
+    def __data__(self):
+        return {"element": self.element, "text": self.text}
+
+    @property
+    def text(self):
+        if self.user_text:
+            return self.user_text
+        return "{}_{}_{}".format(self.element.__class__.__name__, self.element.key, self.element.attributes.get("category") if self.element.attributes.get("category") else "")
+
+    @classmethod
+    def from_element(cls, element, attributes=[], base_string="", char_to_replace=None):
```

Comment:
immutable objects (`[]`) should not be set as default value as these are only evaluated once